### PR TITLE
Merging

### PR DESCRIPTION
**Pull request by IlliaDmitriiev**
- merged storage by priority into view by priority
- merged storage by date into view by date
- deleted storage by priority, storage by date and tests on these classes
- updated tests of views
- changed return type in all views of function: deleteTask

### CODE REVIEW

File: src/logic/ViewByDate.cpp [hunk 1]
```diff
@@ -7,52 +7,67 @@
...
     for (int i = 1; i<=7; ++i) {
         int WeekDay = date.getWday();
         if (WeekDay == 0) WeekDay = 7;
         date.moveWithinTheWeek(i - WeekDay);
 
-        auto NeedData = map.find(date);
-        if (NeedData != map.end()) {
+        auto NeedData = map_.find(date);
+        if (NeedData != map_.end()) {
             auto PriorityMap = NeedData->second.getMap();
```
Comment (denys-hrulov) [inline src/logic/ViewByDate.cpp]: Must be:

const auto& PriorityMap = NeedData->second.getMap();

Comment (IlliaDmitriiev) [reply src/logic/ViewByDate.cpp]: completely agree with you

File: src/logic/ViewByDate.cpp [hunk 2]
```diff
@@ -7,52 +7,67 @@
...
 
-            for( auto i: PriorityMap[1])
+            for( auto i: PriorityMap[Task::Priority::First])
```
Comment (denys-hrulov) [inline src/logic/ViewByDate.cpp]: If you have const issues, write following:
PriorityMap.at(Task::Priority::First)
and in other place the same

Comment (IlliaDmitriiev) [reply src/logic/ViewByDate.cpp]: ok, thanks

**Review by denys-hrulov — APPROVED** [review]

**Pull request merged**